massa-models/src/operation.rs

```diff
@@ -972,13 +970,13 @@ impl SecureShareOperation {
     }
 
     /// Get the max amount of gas used by the operation (`max_gas`)
-    pub fn get_gas_usage(&self) -> u64 {
+    pub fn get_gas_usage(&self, gas_usage: u64) -> u64 {
         match &self.content.op {
             OperationType::ExecuteSC { max_gas, .. } => *max_gas,
             OperationType::CallSC { max_gas, .. } => *max_gas,
```

Comment:
also apply `base_op_gas_usage` to executesc and callsc: `max_gas.saturating_add(base_op_gas_usage)`

Reply:
even better:

```
match ... {
   executesc => max_gas,
   callsc => max_gas
   rollbuy => 0
   rollsell => 0
   transaction => 0
}.saturating_add(base_op_gas_usage)
```

Reply:
done.
@Eitu33: any conflict with your work here?

Reply:
Not really, max_gas will be the same for OP and ABI and base_op_gas_usage will even out abi_cost. The total cost won't be the exact same but it's not an important difference at all.

Reply:
Great, thanks